pkg/sdk/tasks_def.go

```diff
@@ -1,52 +1,130 @@
 package sdk
 
-import g "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/poc/generator"
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	g "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/poc/generator"
+)
 
 //go:generate go run ./poc/main.go
 
+type TaskState string
+
+const (
+	TaskStateStarted   TaskState = "started"
+	TaskStateSuspended TaskState = "suspended"
+)
+
+func ToTaskState(s string) (TaskState, error) {
+	switch taskState := TaskState(strings.ToLower(s)); taskState {
+	case TaskStateStarted, TaskStateSuspended:
+		return taskState, nil
+	default:
+		return "", fmt.Errorf("unknown task state: %s", s)
+	}
+}
+
+type TaskRelationsRepresentation struct {
+	Predecessors  []string `json:"Predecessors"`
+	FinalizerTask string   `json:"FinalizerTask"`
```

Comment:
Nit: If this JSON uses upper case names, these tags can be dropped.

Reply:
Yeah, I know, but I guess this makes it more clear what the struct is used for and it was just two additional tags.